### ISSUE TITLE
Event de-structuring is cumbersome.

### ISSUE THREAD

**Issue by luukvanderduim** [odilia-app/atspi#37]
Currently the event de-structuring looks as such:
```Rust 
while let Ok(ev) = event_stream.next().await.expect("None") {
    let Event::Interfaces(EventInterfaces::Object( ObjectEvents::StateChanged(state_ev))) = ev else { continue; };

    if state_ev.state() == "focused" {
        let Some(sender) = state_ev.sender()? else { continue; };
        println!("Focused by: {sender}")
     }
}
```
If `Event` would hold a copy of `AtspiEvent` (instead of `EventInterfaces`) and the `TryFrom` implementations are adjusted to convert from `Event` to respectively: the signified types or the grouped `EventInterfaces`, we could:
 
```Rust
while let Ok(ev) = event_stream.next().await.expect("None") {
    let ev: StateChangedEvent = ev.try_into()? else { continue; };

    if ev.state() == "focused" {
        let Some(sender) = state_ev.sender()? else { continue; };
        println!("Focused by: {sender}")
     }
}
```
perhaps have current behaviour as such:

```Rust
while let Ok(ev) = event_stream.next().await.expect("None") {
    let evs: EventInterfaces = ev.try_into()? else { continue; };

    /// Then do destructuring ...
}
```

**Comment by TTWNO** [comment]
It may be cumbersome, but it is very easily typed and has a good structure.

If you want a way to do this with AtspiEvent, then why not with Cache and Availsble events too? And at that point, what's the use of the enum?

I think a good comprimise would be to have *all* event variant in Event itself, without splitting them into individual, progressively more specific, enums. Either work well for Odilia.

I'll ping @DataTriny and see what they think.

**Comment by TTWNO** [comment]
Resolved by #50
